app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -465,13 +465,16 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     )
 
     /**
-     * Caches the last known "is default browser" state when the app was paused.
-     * For an up to do date state use `isDefaultBrowser` instead.
+     * Declared as a function for performance purposes. When this class is first initialized, the
+     * isDefaultBrowser function takes approx. 30-40ms.
      */
-    var wasDefaultBrowserOnLastResume by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_default_browser),
-        default = isDefaultBrowser()
-    )
+    fun checkDefaultBrowserAndSet(): Boolean {
+        val prefKey = appContext.getPreferenceKey(R.string.pref_key_default_browser)
+        val isDefaultBrowserNow = isDefaultBrowser()
+        val isDefaultBrowserBefore = this.preferences.getBoolean(prefKey, isDefaultBrowserNow)
+        this.preferences.edit().putBoolean(prefKey, isDefaultBrowserNow).apply()
+        return isDefaultBrowserNow != isDefaultBrowserBefore
```

Comment:
**issue:** this is a change to the original logic.

**suggestion (blocking):** let's functionally restore the original logic. The original logic is:
```kotlin
isDefaultBrowser &&
    wasDefaultBrowserOnLastResume != isDefaultBrowser
```

Tbf, the original logic is a little confusing. I think it'd be clearer as:
```kotlin
isDefaultBrowser && !wasDefaultBrowserOnLastResume
```

i.e. what we're trying to determine is if the user has switched from fenix not being their default browser to fenix being their default browser. You could keep the original logic (less likely to introduce bugs) or use something clearer.

Reply:
I'm not sure how this changes the logic ( I probably misunderstood).

The way I understood it was `wasDefaultBrowserOnLastResume` fetched from the shared preference with `booleanPreference`  whereas now this `wasDefaultBrowserOnLastResume` is replaced by  `val isDefaultBrowserBefore = this.preferences.getBoolean(prefKey, isDefaultBrowserNow)` which does the same same as `by booleanPrefreference`, just calling it from the sharedpreference class instead.  The default value is still the same (`isDefaultBrowser`)

we used to have `settings().wasDefaultBrowserOnLastResume != settings().isDefaultBrowser()`. In this case,  `settings().wasDefaultBrowserOnLastResume` is replaced  by `isDefaultBrowserBefore` and `settings().isDefaultBrowser()` is replaced by the
`val isDefaultBrowserNow = isDefaultBrowser()`.

Reply:
I think I'll just simplify it to the last expression you gave. It'll avoid confusion I think of having 3 variables in the return statement